### PR TITLE
Refactoring SARIF description fields and their respective spec files

### PR DESCRIPTION
**Pull request by saron-deb**
**Jira Ticket:**
SECT-755

**What changed? Why?**
Breaking out SARIF description fields using SARIF supported rule.reportingDescriptor.messageStrings field. This is being done in order to allow more flexibility when displaying and augmenting the description fields in Salus bot. 

**How has it been tested?**
Manual testing

**Change management**

type= routine
risk=low
impact= 

automerge=

### CODE REVIEW

File: spec/lib/sarif/bundle_audit_spec.rb [hunk 1]
```diff
@@ -32,12 +32,13 @@
 
       it 'parses information correctly' do
         bundle_audit_sarif = Sarif::BundleAuditSarif.new(scanner.report)
-        issue = scanner.report.to_h[:info][:vulnerabilities][0]
+        issue = scanner.report.to_h[:info][:vulnerabilities]
+        issue = issue.detect { |i| i[:cve] == 'CVE-2021-22885' }
```
Comment (joshua-ostrom) [inline spec/lib/sarif/bundle_audit_spec.rb:36]: Nice job

File: spec/lib/sarif/bundle_audit_spec.rb [hunk 2]
```diff
@@ -84,36 +85,44 @@
       it 'should return valid sarif report' do
         report = Salus::Report.new(project_name: "Neon Genesis")
         report.add_scan_report(scanner.report, required: false)
-        result = JSON.parse(report.to_sarif)["runs"][0]["results"][0]
-        rules = JSON.parse(report.to_sarif)["runs"][0]["tool"]["driver"]["rules"][0]
 
-        if rules['id'] == 'CVE-2021-22885'
+        results = JSON.parse(report.to_sarif)["runs"][0]["results"]
+        result = results.detect { |r| r["ruleId"] == 'CVE-2021-22885' }
```
Comment (joshua-ostrom) [inline spec/lib/sarif/bundle_audit_spec.rb:90]: since you reference it twice might be nice to store in a var

Comment (joshua-ostrom) [reply spec/lib/sarif/bundle_audit_spec.rb:90]: (CVE-2021-22885)

Comment (saron-deb) [reply spec/lib/sarif/bundle_audit_spec.rb]: Fixed, thanks!

**Review by joshua-ostrom — APPROVED** [review]

**Pull request merged**